sunflower-land/sunflower-land
# [FEAT] Markdown for Announcements

# Description

This PR implements markdown in Announcements feature. Uses `react-markdown` and `remark-gfm` packages. Primary use case will be for links e.g. Discord, Twitter, etc. This does have its limitations as observed. See screenshots in Testing section.

1. Links DO NOT open in new tab
2. Bold / Strong markdown makes the text blurry
3. No underline support
4. If there is no title and first 10 characters do not complete a markdown, it renders as string.

Fixes #issue

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Manual testing

Steps
1. [optional] clear `readMessages` in localStorage
2. Load the game
3. Click bald man and read all messages -- should see bold, italic, strikethrough, and links
4. Click on a link -- will redirect to the link (no open in new tab support)

https://user-images.githubusercontent.com/89294757/155938780-9fc5a22c-d388-4175-8893-c6b84b4e875b.mp4

Blurry bold text
![sfl-markdown-bold](https://user-images.githubusercontent.com/89294757/155938855-3f36fb57-ce7c-4795-aea4-e699b6f5768a.PNG)

No title and starting characters have incomplete markdown
![sfl-markdown-incomplete](https://user-images.githubusercontent.com/89294757/155938970-9ff3241a-059c-4ea6-a81f-9ae4da9ecbca.PNG)

`yarn tsc|test|format|audit` showed no errors related to code/packages used.

# Checklist:
